### ISSUE TITLE
[java] New rule: UnusedReturnValue

### ISSUE THREAD

**Issue by UncleOwen** [pmd/pmd#6625]
**Proposed Rule Name:** UnusedReturnValue (~CheckReturnValue~)

**Proposed Category:** Error Prone

**Description:** A lot of libraries nowadays include a `CheckReturnValue` annotation: [errorprone](https://errorprone.info/bugpattern/CheckReturnValue), [spring](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/lang/CheckReturnValue.html), [findbugs](https://www.javadoc.io/doc/com.google.code.findbugs/jsr305/latest/javax/annotation/CheckReturnValue.html), [intelliJ](https://github.com/JetBrains/java-annotations/blob/master/src/jvmMain/java/org/jetbrains/annotations/CheckReturnValue.java)

Meaning is the same every time:

- If on a method, you should use the return value of this method somehow
- If on a class/package, you should treat every method in this class/package as if it were annotated with `CheckReturnValue`, unless it's annotated with `CanIgnoreReturnValue`

The new rule should check this. It's just the name of the annotation that is relevant, the package name can be ignored.

**Code Sample:** 
Bad:
```java
    logger.atError().setMessage("some message with two arguments {} ms: {}")
          .addArgument("some string argument")
          .addArgument(() -> System.currentTimeMillis() - startDate);
```

Good:
```java
    logger.atError().setMessage("some message with two arguments {} ms: {}")
          .addArgument("some string argument")
          .addArgument(() -> System.currentTimeMillis() - startDate)
          .log();
```
(Examples copied from https://github.com/pmd/pmd/discussions/5063)

**Possible Properties:**

*   We should have a property where the user can define additional `InvocationMatcher`s. The return value of these calls should also be checked. (Maybe we can copy the list of methods from IntelliJ's "Result of method call ignored" inspection as a default for additional methods to check? In that case, this rule would subsume [UselessPureMethodCall](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#uselesspuremethodcall), [CheckSkipResult](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#checkskipresult), [CheckResultSet](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#checkresultset) and possibly others.)

**Comment by UncleOwen** [comment]
Inspectopedia: [ResultOfMethodCallIgnored](https://www.jetbrains.com/help/inspectopedia/ResultOfMethodCallIgnored.html)
